### PR TITLE
Add EDD Wallet support

### PR DESCRIPTION
**Pull request by ghost1227**
This adds support for the EDD Wallet extension, and converts the plugin to the new AffWP multi-integration format.

### CODE REVIEW

**Comment by ghost1227** [comment]
I should note that while I did add the updater call present in the sample you provided, I don't know the item ID for Store Credit, so right now it's listed as '0000'. This will have to be updated.

**Comment by pippinsplugins** [comment]
@ghost1227 No need for the updater. This is a free add-on that's hosted on WordPress.org. Could you remove  that?

**Comment by ghost1227** [comment]
Done

File: integrations/class-base.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+abstract class AffiliateWP_Store_Credit_Base {
+	public $context;
+
+	public function __construct() {
+		$this->init();
+		$this->setup();
+	}
+
+	public function init() {
```
Comment (pippinsplugins) [inline integrations/class-base.php]: Let's combine init and setup

File: integrations/class-edd.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	protected function add_payment( $referral_id ) {
+
+		// Return if the referral ID isn't valid
+		if( ! is_numeric( $referral_id ) ) {
+			return;
+		}
+
+		// Get the referral object
+		$referral = affwp_get_referral( $referral_id );
+
```
Comment (pippinsplugins) [inline integrations/class-edd.php]: This should only run if the referral is set to "paid"

**Comment by ghost1227** [comment]
Done

**Pull request merged**